internal/policy/operator/deployable_by_olm.go

```diff
@@ -248,11 +251,19 @@ func (p *DeployableByOlmCheck) operatorMetadata(ctx context.Context, bundleRef i
 		installModes[val.Type] = val
 	}
 
+	// validating that package name complies with DNS-1035 labeling constraints
+	// ensuring that CatalogSources can be created/referenced in cluster
+	appName := packageName
+	if msgs := validation.IsDNS1035Label(packageName); len(msgs) != 0 {
+		logger.V(log.DBG).Info(fmt.Sprintf("package name %s does not comply with DNS-1035, prefixing to avoid errors", packageName))
+		appName = "preflight-" + packageName
```

Comment:
Why are we doing this instead of just failing this check here?

Reply:
We need the operator to still be able to be installed to avoid this error being thrown by k8s further along in the process.

Reply:
I'd go for a shorter prefix... `pflt-` or even `p-`. You don't want to possibly hit a long one and end up failing because it goes beyond 63.

Reply:
Either:
a) Make this prefix shorter and validate that it fixes the problem before use (by re-running the isDNS!035Label func again on the new value), or
b) Generate the name used to create the catalog source in a way that's reliably formed.